src/js/resolver.js

```diff
@@ -14,9 +14,28 @@ Resolver.prototype.resolve = function(spec, callback, scope) {
   // models
   for(name in spec.definitions) {
     var model = spec.definitions[name];
-    for(propertyName in model.properties) {
-      property = model.properties[propertyName];
-      this.resolveTo(property, resolutionTable);
+    if (!model.properties) {
+      model.properties = {};
+    }
+    if (model.allOf) {
+      for(mergeName in model.allOf){
```

Comment:
@wodka I misread your diff. Sorry about that. You don't have to change the code you didn't write but if you refactor those parts it would be nice.